### ISSUE TITLE
Beta testing

### ISSUE THREAD

**Issue by MSghais** [AFK-AlignedFamKernel/afk_monorepo#2]
** Stay tuned for the announcement and the start of the Beta testing**

# Description
The beta testing contest starts soon. The date for start and end of the contest is announced at the same time.

You need to send your report before the end date, as described in the next steps.

How to test and help us
The demo is live here:
https://afk-community.xyz/

You can request to be assigned to the GitHub issue.
Send a Post (Note) on AFK with your GitHub ID (check this step to get your Github ID: https://t.me/JoyboyStarknet/1/2864
Come back later, and if you are assigned you can start the beta test:
Let's write an issue or a doc with a description of what you did, as well as your Nostr profile and Starknet address.

# Submit Report
Here is the description to send your report:

Report any problems you encounter in just a few steps:
What the problem is, how to reproduce the steps, error logs, and messages that may help.
You can open different issues for these (check if it's not already reported maybe)

## Starknet test
Transaction hash for :

Giving a tip to a friend.

- [ ] Initial claim (without funds, sent by payer)
- [ ] Claim a deposit (sign the transaction with your wallet).
- [ ] Create token
- [ ] Launch token
- [ ] Buy token
- [ ] Sell token

## View app and Screens :
Your GitHub ID comment is here in a comment, and on AFK.
- [ ] Login/Sign or use extension
- [] Save your Private key of Nostr
- [ ] Edit profile screens with your sent notes and your profile on AFK.
- [ ] Onboard

- [ ] Profile of users
- [ ] Post detail
- [ ] Create a Nostr account to login, in with a username and a password.
- [ ] Channels
- [ ] Search bar

Go to the home page and visit the app (Feed of notes, your profile, others profiles)

### Nostr

- [ ] Send a note:
- [ ] Send AM with your GitHub ID (not required), and request a tip with some thoughts, ideas, or feedback about AFK check this step to get your Github ID: https://t.me/JoyboyStarknet/1/2864)

- [ ] Comment a note/comment of another user
- [ ] Give a reaction to a Note
- [ ] Edit your profile
- [ ] Click to the profile icon
- [ ] Add your Username and NIP-5

### Tipping task
Create a Starknet Wallet in your Browser extension: Argent, Bravoos, OKX.
You can install one directly on AFK:

#### Create your wallet
Save your Starknet private key somewhere

Request fun on AFK:
Send a note: JM please tips me.
One user or the team gonna send you a tip.

Come back a few moment later:
Go to the claim page as described below and claim your first fund paid by the paymaster.

Login with your Wallet (Argent supported on Mobile, Bravoos OKX etc on Web view)
Send a tip to your friends
Click on the icon of a Notes or in a User profile

image

Click on Tips. Send the Token and the amount you want to used.

image

Connect your wallet

Sign the transaction to tip your friends in your wallet.

Get confirmation of your tips is sent correctly

#### Claim your tips
Go to the Claim page (Click on the 2th button in the Bottom bar)
If you don't have tips, you can tip you or:
Request a tip on AFK:
Send a note: JM please tips me.
One user or the team gonna send you a tip.
Go to the claim a few moments after (we are reactive)

### Select a Tips to claim (if not yet done)
If you don't have tips to claim, send a tip directly to you, or request a tip in the Telegram or on X with your Nostr public key

### Claim your tips

Click on the claim button.
How it work:

If you have some ETH or Starknet fund: you can sign the tx to claim your fund
If no ETH fund, the paymaster claims it for you.
image

Transaction hash succeeds.

You see the label Claimed with a Disabled button

Give your feedback and report the bug
Here is the description (present in the beta test issue) to send your report:

Report any problems you encounter in just a few steps:
What the problem is, how to reproduce the steps, error logs, and messages that may help.
You can open different issues for these (check if it's not already reported maybe)

Transaction hash for :

Giving a tip to a friend.
Initial claim (without funds, sent by payer)
Claim a deposit (sign the transaction with your wallet).
Screens :
Your GitHub ID comment is here in a comment, and on AFK.
Edit profile screens with your sent notes and your profile on AFK.
Feedback:

What do you think, you want, you need? Let's discuss it!

**Comment by Joy-Adah** [comment]
Hi @MSghais I'll love to participate in the beta testing. Can I be assigned?

**Comment by MSghais** [comment]
> Hi @MSghais I'll love to participate in the beta testing. Can I be assigned?

HI @Joy-Adah, the beta test has not started yet.
To stay updated, you can join the AFK Telegram group and follow us on X!

Telegram: https://t.me/afk_aligned_fam_kernel

X: https://x.com/AFK_AlignedFamK/status/1820153289801187504

A lot of cool things are coming

**Comment by raizo07** [comment]
Hello @MSghais I'll love to participate.

**Comment by web3g1rl** [comment]
Hi @MSghais I'll love to participate in this.

**Comment by od-hunter** [comment]
Hello @MSghais , I’d love to participate please.

**Comment by ShantelPeters** [comment]
Hello @MSghais I’ll love to participate please.

**Comment by Oshioke-Salaki** [comment]
Hello I would like to participate

**Comment by Jemiiah** [comment]
Hello @MSghais i would love to participate in this

**Comment by martinvibes** [comment]
hello @MSghais i would also love to participate if i'm giving the oppurtunity
i used to be a QA tester in [test.io](https://test.io/)

**Comment by petersdt** [comment]
@MSghais can i participate in this project

**Comment by MSghais** [comment]
Gm everyone! Thank you for applying!

We are going to postpone this issue for a few days or 2 weeks max, and we announce when it's ready!
Cool features etc to test in a few :)

Stay tuned and join the Telegram group to stay up to date:
https://t.me/afk_aligned_fam_kernel

**Comment by lauchaves** [comment]
Looking forward to this! 🚀

**Comment by CollinsC1O** [comment]
Hello @MSghais I will really love to participate on this

**Comment by GoSTEAN** [comment]
@MSghais Can I work on this issue?

**Comment by MSghais** [comment]
Gm everyone, the OD Hack on Only Dust is starting Thursday!

A lot of issues are incoming, stay tuned and join the TG group:
https://t.me/afk_aligned_fam_kernel

**Comment by petersdt** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I am Peter a frontend developer with experience in writing test for frontend applications and programs. With some level of experience in application testing

### How I plan on tackling this issue
1. Create Required Accounts using username

**Comment by martinvibes** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
i'm a frontend developer with experience o testing for a quality assurance testing company called test.io

### How I plan on tackling this issue
Nostr Account: Create a Nostr account with a username and password. Ensure the private key is saved securely.
Starknet Wallet: Install a Starknet-compatible wallet (e.g., Argent, Bravoos, OKX) and save the private key.
AFK Account: Log in to AFK using the Nostr account and set up the profile.

**Comment by od-hunter** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

**My background and how it can be leveraged**

i'm a blockchain developer and I have experience in beta testing. I was part of Joyboy beta testing, I would love to be given the opportunity to test and document here please.

How I plan on tackling this issue
I’d login to test and see if each featured function properly, and if they don’t I document down and report.

**Comment by NueloSE** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I am a frontend developer with expierence in write test for frontend applications.
Once worked with a testing platform called test.io . I believe with my level of experience from these i would be able to check for bugs during the beta testing

**Comment by GoSTEAN** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I have a solid background in web development, blockchain technology, and decentralized applications (dApps). My experience in building scalable, secure, and user-friendly web platforms, combined with a deep understanding of smart contracts, enables me to effectively test and provide valuable feedback on AFK's functionality. I can leverage my skills to identify potential issues, suggest improvements, and ensure the application meets high standards of performance and usability.

### How I plan on tackling this issue

**How Would You Approach the Problem?**

1. **Understanding the Requirements:**
   - Carefully review the beta testing guidelines, objectives, and expected outcomes.
   - Ensure I have all necessary tools and accounts set up (e.g., GitHub, Nostr, Starknet wallet) before starting.

2. **Testing the Features:**
   - Methodically go through each task listed, from sending a note to tipping friends, and document any issues encountered.
   - Pay attention to edge cases, user experience, and overall app performance.

3. **Reporting Issues:**
   - For each issue, provide a clear and concise report, including steps to reproduce, error logs, and screenshots.
   - Prioritize issues based on severity and impact on user experience.

4. **Providing Feedback:**
   - Offer constructive feedback on the overall design, functionality, and potential areas for improvement.
   - Suggest enhancements based on my experience and understanding of user needs.

5. **Collaboration:**
   - Actively engage with the community and development team, discussing findings and potential solutions.
   - Stay involved throughout the beta testing process, ready to test new updates and provide further feedback as needed.

**Comment by lauchaves** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Hey! I'm Lau Chaves, and I would like to contribute to this issue!

I have over 5 years of experience working with JavaScript, React, and TypeScript, ruby... Im also part of Dojo Coding Costa Rica!

Feel free to check my onlydust profile here and github profile: lauchaves!


### How I plan on tackling this issue
I have some experience with beta testing, having participated in one before!

Here’s what I’d do:

- Play around with the AFK demo to spot any user experience issues.
- Review the local code and look for improvements, comparing it with the live demo.
- Document any feedback or suggestions I come across.
- Help create issues on GitHub if needed and might even contribute to addressing them.

I also have strong writing skills to clearly document findings and suggestions.

---

**Comment by lindsaymoralesb** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Hi! My name is Lindsay, and I'm a Software Engineer based in Costa Rica. Also part of Dojo Coding.
I've been working over the past 4.5 years as a full stack developer, specializing myself in frontend development. I also have some qa experience with both manual testing and automation.

### How I plan on tackling this issue
I contributed to the Beta Testing on JoyBoy app and was able to spot some issues. What I would do is:
1. I already posted my gh id on the app demo. 
2. Test the AFK demo thoroughly to identify any user experience or behavioral issues.
3. Document any issues or improvement suggestions I encounter.
4. Share the issues evidence and steps to reproduce on GitHub issues.
5. If possible, I would like to help by solving the accepted issues by the team.

**Comment by MullerTheScientist** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
My Background and How It Can Be Leveraged:
 I possess comprehensive expertise as a front-end developer coupled with a solid foundation in QA testing. Presently, I am engaged as an independent tester with Utest Academy. Given my substantial experience, I am confident in my ability to meticulously identify and address bugs during the beta testing phase.

### How I plan on tackling this issue
1. Executing tests, documenting results, and reporting defects.
2. Verifying fixes and re-testing to ensure issues are resolved

**Comment by Supa-mega** [comment]
hi @MSghais i am a front end developer with 10 years experience and will love to work on this project

**Comment by od-hunter** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Hi, I am a blockchain developer and I am also good with beta testing. I was part of the beta testers for JoyBoy and I'd also love to take part in this beta testing.

### How I plan on tackling this issue
I would make sure to for through the site and observe how it responds to things careful, document issues and give honest feedbacks. Please I'd love to participate in this.

**Comment by Ugo-X** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I am a Full Stack blockchain Developer with expertise in Next.js, Nest.js, TypeScript, JavaScript, React, Node.js, Three.js, and Solidity. My journey with OnlyDust hackathons began at Edition 1, and I've since made 47 contributions across 11 projects. With my extensive experience on the OnlyDust platform (profile:  https://app.onlydust.com/u/Ugo-X), I've honed my skills in delivering quality solutions under pressure.
I bring a unique blend of technical proficiency and user-focused design to every project, whether it's crafting immersive 3D experiences or developing smart contracts. My track record shows I can adapt quickly and contribute effectively to diverse challenges.
As we surf through Edition 7, I'm excited to leverage my skills and hackathon experience to push the boundaries of blockchain development. I'm confident in my ability to tackle new challenges and drive innovation in this space.


### How I plan on tackling this issue
. Here's how I would approach the beta testing and reporting for this issue:

1. **Set Up the Necessary Accounts and Tools**:
   - Ensure I have a GitHub account and can access the relevant issue or repository.
   - Create a Nostr profile and obtain my Nostr public key.
   - Set up a compatible Starknet wallet (e.g., Argent, Bravoos, OKX) and securely store the private key.

2. **Familiarize Myself with the AFK Community App**:
   - Thoroughly explore the live demo available at https://afk-community.xyz/.
   - Understand the key features and functionality, including the Nostr integration and Starknet-based transactions.

3. **Methodically Test the Specified Tasks**:
   - Follow the steps outlined in the beta testing instructions, systematically going through each task.
   - Document any issues or problems encountered, including:
     - Detailed description of the problem
     - Steps to reproduce the issue
     - Any error logs or messages
     - Screenshots or recordings to help illustrate the problem

4. **Report Issues and Provide Feedback**:
   - For each issue identified, create a new GitHub issue or document it in the provided format.
   - Include the relevant transaction hashes, Starknet addresses, and Nostr profile information as requested.
   - Provide constructive feedback on the overall user experience, design, and potential areas for improvement.

5. **Engage with the Community and Development Team**:
   - Actively participate in the beta testing community, such as the Telegram group or Nostr discussions.
   - Share my findings, insights, and suggestions with the team and other beta testers.
   - Be open to testing new updates or features as they become available during the beta period.

6. **Ensure Comprehensive Testing and Reporting**:
   - Thoroughly test all the specified tasks, including edge cases and unexpected user inputs.
   - Provide detailed and well-structured reports for each issue, making it easy for the development team to understand and address them.
   - Offer thoughtful feedback and ideas for improving the AFK Community app based on my testing experience.

By following this approach, I will ensure a thorough and effective beta testing process, providing valuable insights and feedback to the development team to help enhance the AFK Community app.

**Comment by Joy-Adah** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Hi! My name is Joy and I am a developer. I have experience in testing apps.

### How I plan on tackling this issue
I have already created my profile on AFK and I plan to play around the app, test all functionalities and document my feedback

**Comment by web3g1rl** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Hello! I am a frontend developer that just joint the onlydust platform. I have experiences when it comes to testing of applications.

### How I plan on tackling this issue
I already have a profile created on AFK and I intend to try and test out all functions on the application, take screenshots of all process and document all my findings appropriately.

**Comment by MartinXV111** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I'm not new in the beta testing field, i did one for a Starknet social network similar to yours not long ago and that went pretty well. I'm sure I'll be able to find necessary bugs and issues and document them as the project description demands.

### How I plan on tackling this issue
I've already started to take down issues/bugs that I've noticed on the app as it's not my first time using the app, I've been on it for quite some time now so I'll continue testing and reporting bugs as the description of the issue demands.

**Comment by MSghais** [comment]
Gm everyone!

Please use the Only Dust platform to apply for this issue.

**Comment by martinvibes** [comment]
hello @MSghais should we start testing?

**Comment by MartinXV111** [comment]
Hey, here's my GitHub ID :)
112659651

**Comment by MartinXV111** [comment]
[Report Submission For AFK Beta Test.docx](https://github.com/user-attachments/files/16830727/Report.Submission.For.AFK.Beta.Test.docx)
Here's my report for the Beta Test :)

**Comment by pheobeayo** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I am a web3 frontend developer and Technical writer skilled in Typescript, Javascript, Solidity, Cairo and Rust.

### How I plan on tackling this issue
I will do this documentation for beta testing in the following steps
- I will the app successfully locally.
- I will document every step as stated in the requirement.
-I will test every function on the app and document the all steps involved in that.

**Comment by gregemax** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I have a solid background in software development, particularly in Go, JavaScript, and Python, with hands-on experience in testing and debugging decentralized applications. My familiarity with smart contracts equips me to effectively evaluate and report issues in blockchain systems, ensuring a seamless user experience

### How I plan on tackling this issue
I would begin by exploring the demo at https://afk-community.xyz/ to understand its functionalities. Then, I would set up my Starknet wallet and systematically test each feature, documenting any issues along with reproduction steps and error messages. Finally, I would compile a comprehensive report with my findings and suggestions, ensuring clear communication for effective improvements.

This structured approach will help identify and address issues efficiently, contributing to a polished final product.

**Comment by deeseeker** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Hi, I'm Qudus Adeyemi, a front-end developer with 2 years of experience. I had the opportunity to contribute to the previous OD Hackathon, where I improved the Starknet React documentation and developed a useWatchAssets hook to enhance wallet tracking functionalities. I enjoy learning and tackling new challenges, and I see this issue as a great way to deepen my understanding of Web3 technologies while contributing to the AFK project. I'm excited to dive into the task and collaborate with the team!

### How I plan on tackling this issue
This is my first time doing a test so I don't have any plan aside from following the steps, researching and communicating with the team. Though, this is my first time doing a test, I am not afraid to take on the challenge as I did previously with Starknet React.

**Comment by Nemezjusz** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I bring a fresh perspective to the AFK platform as a developer starting his journey in blockchain. I have experience writing web apps and improving user experinece. I also work as a tester in telecommunication.

### How I plan on tackling this issue
As a tester, I like to look into various edge cases but also try to test how intuitive interface is. I yhink my perspective  is valuable because it represents how first-time users might interact with the platform.

**Comment by suhas-sensei** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
Whew, looks like a ton of applications up there. Working as a Meme-Driven Dev back at Gobblchain, i think i can put my learnt skills at work.

with my previous experiences in personal projects and hackathons, I can whip up some real solutions—probably with a one-liner or two and a laugh along the way. [it's my first ODHack :)]


### How I plan on tackling this issue
ok so,
1. I will set up the Nostr account and save my private key. Then link it with the Braavos wallet.
2. I'm gonna make sure that all the functions work, explore every possible scenario and make a note of the issues, if any.
3. Check if the UX needs some work or is fine. I'll test out tipping, request some tips, claim them, and flex my wallet setup by sending tokens.
4. I'll document everything if i find out something fishy, which will include steps to recreate, error logs, screenshots and other stuff. I'll make sure the team is going to love my clarity.
5. Provide feedback from my end too, as a dev :)

Looking forward.

**Comment by ShantelPeters** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I am a blockchain developer and i have a little bit of experience in beta testing and a strong background experience working with React, ,javascript, typescript , cario and solidity.
i have contributed successfully to open source projects like Arkprojects and Shinigami .

### How I plan on tackling this issue
I will approach thee problem by doing a documentation for beta testing by the following
By taking down bugs that i notice on the app 
i will try and test out all functions on the application ,take screenshots and approiately document all my findings.
Pay attention to user experience and overall app performance 
I will also ensure i have necessary tools and account set up e.g Nostr, starknetwallet, Github. 
Looking forward .....

**Comment by MartinXV111** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I have a very strong background with running beta tests especially for social networks built on nostr. I have worked with joyboy and Also AFK in the past. In the long concluded OD Hack 7.0 i was assigned the task of beta test for AFK and did the job as well as i could, i was also rewarded for this task.

### How I plan on tackling this issue
I'll approach the problem by first making myself acquainted with the social network, which I've already said before that I'm well familiar with it (AFK that is). I'll then look for bugs and issues i encounter while using the platform and I'll also follow the guided prompts asked by the maintainer :)

**Comment by Damilola145** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I'm a front-end developer with a nack for trying new things.I love taking on challenges as it spurs the intuitive part of me to produce creative solutions.This will aid me during the testing to highlight any problem as observed during the testing period

### How I plan on tackling this issue
I will make sure to follow the procedures as highlighted in line with the testing,and also make appropriate reports based on the results gotten from the testing period

**Comment by gregemax** [comment]
Hi @MSghais 
I will like to be assign this issues

**Comment by obacollins-lab** [comment]
I am applying to this issue via [OnlyDust platform](https://app.onlydust.com/p/afk---aligned-fam-kernel).

### My background and how it can be leveraged
I am a software enthusiast with experience in testing, debugging, and contributing to open-source projects. My background in blockchain technology and user experience (UX) analysis equips me to provide meaningful feedback for decentralized applications. I have previously tested several blockchain-based projects and reported bugs, suggested UX improvements, and collaborated with development teams.

### How I plan on tackling this issue
1. Understand Requirements
Review the testing scope and contest guidelines to identify key areas and features to test.

2. Set Up Environment
Create a Starknet wallet, set up profiles, and ensure access to the demo and GitHub issue tracker.

3. Test and Document
Test all major scenarios, including tipping, claiming, and interacting with features. Document issues, providing detailed reproduction steps and logs.

4. Verify Transactions
Execute all transactions, log hashes, and report any issues or unexpected behavior.

5. Feedback and Suggestions
Provide constructive feedback on UX and potential improvements, along with issue reporting.

This approach ensures thorough testing and meaningful contributions to the AFK Community project. Looking forward to participating!
